drivers/vmwarevsphere/cloudinit.go

```diff
@@ -228,59 +230,118 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
 	return vm.AddDevice(d.getCtx(), add...)
 }
 
-func addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
-	cf := make(map[interface{}]interface{})
-	if err := yaml.Unmarshal([]byte(yamlcontent), &cf); err != nil {
-		return "", err
-	}
+func (d *Driver) addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
+	switch {
+	case strings.ToLower(d.MachineOS) == defaultMachineOS:
```

Comment:
```suggestion
	switch strings.ToLower(d.MachineOS) {
	case defaultMachineOS:
```

Reply:
Also, a lot of the code in this block is the same as the Windows block. This should be consolidated.